### ISSUE TITLE
Flow errors

### ISSUE THREAD

**Issue by svipben** [stylelint/stylelint#2515]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

I'm using `Flow` with `stylelint` and flow shows me errors in `stylelint` package. If you want to exclude these errors you need to add `stylelint` in `[ignore]`, but I think better solution will be just remove these comments or fix them.

> Which rule, if any, is this issue related to?

Not rule... Flow shows errors in stylelint package, because of bad comments.

> What CSS is needed to reproduce this issue?

Not needed...

> What stylelint configuration is needed to reproduce this issue?

Doesn't matter.

> Which version of stylelint are you using?

7.10.1

> How are you running stylelint: CLI, PostCSS plugin, Node API?

I'm using VS Code [plugin](https://marketplace.visualstudio.com/items?itemName=shinnn.stylelint).

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Nop.

> What did you expect to happen?

Don't show errors, because it's hard to see my own errors in Flow, because of stylelint, number of bad comments which reproduce errors in Flow is 39 (you can see it in picture).

> What actually happened (e.g. what warnings or errors you are getting)?

Everything is explained above and also everything is shown in picture.

![screenshot from 2017-04-22 11-14-52](https://cloud.githubusercontent.com/assets/10441177/25302745/d720e42a-274d-11e7-8c69-7d3f7f741864.png)

**Comment by hudochenkov** [comment]
Please, use issue template.

**Comment by svipben** [comment]
@hudochenkov Done.

**Comment by jeddy3** [comment]
Possibly related to https://github.com/stylelint/stylelint/pull/2471. @svipben Would you like to help resolve this issue by investigating further and ensuring the stylelint code is compatible with the latest flow version?

**Comment by svipben** [comment]
@jeddy3 I'm using Flow 0.44.2 (latest). What I need to-do?

**Comment by jeddy3** [comment]
In the #2471 there are [flow warnings](https://travis-ci.org/stylelint/stylelint/jobs/218515590). That pull request (created by greenkeeper) tries to update flow to the latest version. Perhaps updating the stylelint flow code to be compatible with the latest flow release will help resolve your issue. It at least seems like a good first step.

To help us do this you can:

1. fork this repo
2. run `npm i -D flow-bin@latest` 
3. run `npm run flow`
4. investigate and resolve the warnings
5. submit a PR back to this repo

**Comment by svipben** [comment]
@jeddy3 Done https://github.com/stylelint/stylelint/pull/2524. This is my first pull request, if I did something wrong, say it :smile:

**Comment by jeddy3** [comment]
Done in #2524.

@svipben Thanks again.

**Comment by tirli** [comment]
I'm getting 46 errors in stylelint package using "stylelint": "^8.0.0" and "flow-bin": "^0.52.0",

**Comment by ntwb** [comment]
@tirli I didn't get any errors when I updated `flow-bin` to `0.52.0` in #2793

Ensure your npm modules are all up to date and you are using the latest 5.x release of npm please

Here's my test just now:
```shell
❯ npm run flow

> stylelint@8.0.0 flow /Users/netweb/dev/stylelint/stylelint
> flow

Launching Flow server for /Users/netweb/dev/stylelint/stylelint
Spawned flow server (pid=34844)
Logs will go to /private/tmp/flow/zSUserszSnetwebzSdevzSstylelintzSstylelint.log
No errors!
```

If you still see errors please open a new issue rather than commenting on this old issue please 😄

**Comment by crisu83** [comment]
@ntwb @tirli I still get four Flow errors using **stylelint** ^8.0.0 and **flow-bin** 0.52.0:

```
$ flow
The flow server's version didn't match the client's, so it exited.
Going to launch a new one.

Launching Flow server for /Users/chris/Code/digiaonline/react-boilerplate
Spawned flow server (pid=12026)
Logs will go to /private/tmp/flow/zSUserszSchriszSCodezSdigiaonlinezSreact-boilerplate.log
Error: node_modules/stylefmt/node_modules/stylelint/lib/standalone.js:199
199: function convertCssSyntaxErrorToResult(error/*: CssSyntaxErrorT*/)/*: stylelint$result*/ {
                                                                           ^^^^^^^^^^^^^^^^ property `parseErrors`. Property not found in
              v
204:   return {
205:     source: error.file || "<input css 1>",
206:     deprecations: [],
...:
216:   }
       ^ object literal

Error: node_modules/stylefmt/node_modules/stylelint/lib/utils/getUnitFromValueNode.js:19
 19:   const value = blurInterpolation(node.value, "")
                                                   ^^ unused function argument
                        v------------------------------------------
    3: module.exports = function (source/*: string*/)/*: string*/ {
    4:   const blurChar/*: string*/ = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : " "
    5:
    6:   return source.replace(/[#@{}]+/g, blurChar)
    7: }
       ^ function expects no more than 1 argument. See: node_modules/stylefmt/node_modules/stylelint/lib/utils/blurInterpolation.js:3

Error: node_modules/stylelint-order/node_modules/stylelint/lib/standalone.js:199
199: function convertCssSyntaxErrorToResult(error/*: CssSyntaxErrorT*/)/*: stylelint$result*/ {
                                                                           ^^^^^^^^^^^^^^^^ property `parseErrors`. Property not found in
              v
204:   return {
205:     source: error.file || "<input css 1>",
206:     deprecations: [],
...:
216:   }
       ^ object literal

Error: node_modules/stylelint-order/node_modules/stylelint/lib/utils/getUnitFromValueNode.js:19
 19:   const value = blurInterpolation(node.value, "")
                                                   ^^ unused function argument
                        v------------------------------------------
    3: module.exports = function (source/*: string*/)/*: string*/ {
    4:   const blurChar/*: string*/ = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : " "
    5:
    6:   return source.replace(/[#@{}]+/g, blurChar)
    7: }
       ^ function expects no more than 1 argument. See: node_modules/stylelint-order/node_modules/stylelint/lib/utils/blurInterpolation.js:3


Found 4 errors
```

**Comment by ntwb** [comment]
@crisu83 Those four erros are *not* for stylelint, 2 of each for `stylefmt` and `stylelint-order` in the `node_modules` folder, if you run `npm run flow` it will correctly ignore the `node_modules` folder 😄

**Comment by crisu83** [comment]
AFAIK Flow errors are meant to be fixed instead of being ignored. :)

**Comment by ntwb** [comment]
The errors you see @crisu83 have nothing to do with stylelint as I said above.

**Comment by hudochenkov** [comment]
`stylelint-order` doesn't use flow at all, BTW.

**Comment by crisu83** [comment]
@ntwb So this isn't a bug then in the function definition? Or are you in some kind of transition period?

Usage: https://github.com/stylelint/stylelint/blob/14163e0373b13c95bf7a5f2879ae4d392f4a3085/lib/utils/getUnitFromValueNode.js#L19

Definitions: https://github.com/stylelint/stylelint/blob/14163e0373b13c95bf7a5f2879ae4d392f4a3085/lib/utils/blurInterpolation.js#L4

**Comment by hudochenkov** [comment]
@crisu83 would you mind helping us to fix these errors, please? Currently we don't available maintainers, who know Flow.

**Comment by ntwb** [comment]
What @hudochenkov said, I've very little knowledge of Flow @crisu83 so if this is something stylelint should fix then we're more than happy to fix it, I personally just don't know how 😄
